src/Socket.cpp

```diff
@@ -699,6 +723,15 @@ int Socket::Recv(void* buffer, int length, bool block)
     return recv(socket_, reinterpret_cast<char*>(buffer), length, (block ? 0 : MSG_PEEK) );
 }
 
+int Socket::Recv(void* const buffer, const int length, PeerAddr& addr)
+{
+    if (!isValid())
+        return -1;
+
+    socklen_t addrLen = addr.GetSize();
+    return recvfrom(socket_, reinterpret_cast<char*>(buffer), length, 0, addr.GetAddr(), &addrLen);

```

Comment:
Correct. However I left this to the caller that should notice the error code returned by calling this on a TCP connection. The limitation is also mentioned in the header